tests/5.1/teams/test_target_set_num_teams.c

```diff
@@ -0,0 +1,50 @@
+//===--------------------- test_target_set_num_teams.c ----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the omp_set_num_teams utility to set the default number of
+// teams allocated for team clauses. The test runs teams and checks to see if
+// the number allocated is correct and reflective of the new default set. It
+// then checks to see if an explicitly stated num_teams clause overrides
+// the omp_set_num_teams properly.
+//
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+
+	OMPVV_TEST_OFFLOADING;
+
+	int errors = 0; 
+	int num_teams = 0;
+
+	omp_set_num_teams(1);
```

Comment:
Query max and set it to 1 less than the max value and check from within the target region.

Reply:
I might miss something here, but doesn't the initial _nteams-var_ ICV, which exists per device, have the value zero, and 'omp_set_num_teams' sets the value for the current device (= host)?

Thus, I expect that inside 'omp target teams', the _nteams-var_ ICV is _zero_, unless the default device is the initial device. And if it is zero, regarding the number of teams for a **teams** construct, "the effect is as if _upper-bound_ was specified as an implementation defined value greater than or equal to one."

Or have I missed something? If not:
* One way out would be to test it only on the host (i.e. 'omp teams' not 'omp target teams').
* When using it with 'target', it cannot be in the _same_ target region as 'teams' due to the restrictions on 'teams' inside 'target'.
  However, I wonder whether it would work to set it in one 'omp target' and then use it in another 'omp target teams', relying on the preservation of the ICVs between subsequent target regions.

_(Note: Only glanced at TR11, but I don't expect relevant differences between 5.1, 5.2, and TR11 in this area, except that the handling of env vars such as OMP_NUM_TEAMS was clarified wrt non-initial devices and OMP_NUM_TEAMS_ALL has been added.)_

Reply:
_First, the inconsistency with 'omp_get_max_teams' is now tracked in OpenMP Specification Issue 3619; the sentiment is to change the _default value_ to an implementation defined positive value (currently it is 0), which avoids the issue about what value 'omp_get_max_teams' will report. — But that's as an aside._

Regarding `omp_set_num_teams(1);`, as mentioned, this only sets the ICV of the host. You might want to do:
```suggestion
#pragma omp target
{
	omp_set_num_teams(1);
}
```